src/plasma/plasma_store.cc

```diff
@@ -622,14 +646,16 @@ void send_notifications(event_loop *loop,
   /* Loop over the array of pending notifications and send as many of them as
    * possible. */
   for (int i = 0; i < utarray_len(queue->object_notifications); ++i) {
-    ObjectInfo *notification =
-        (ObjectInfo *) utarray_eltptr(queue->object_notifications, i);
+    uint8_t **notification =
+        (uint8_t **) utarray_eltptr(queue->object_notifications, i);
+    uint8_t *data = *notification;
+    /* Decode the length, which is the first bytes of the message. */
+    int64_t size = *((int64_t *) data);
 
     /* Attempt to send a notification about this object ID. */
-    int nbytes = send(client_sock, (char const *) notification,
-                      sizeof(*notification), 0);
+    int nbytes = send(client_sock, data, size + sizeof(int64_t), 0);
```

Comment:
I have a slight preference for `sizeof(int64_t) + size` since the size is getting sent first.